### PR TITLE
fix Get/PutNoWait deadlock

### PR DESCRIPTION
**Pull request by Arvintian**

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -165,6 +166,7 @@ func (q *Queue) Put(val interface{}, timeout float64) error {
 	q.clearPending()
 	isfull := q.isfull()
 	if timeout < 0.0 && isfull {
+		q.mutex.Unlock()
```
Comment (damnever) [inline queue.go]: Test case does not cover this case.

Comment (Arvintian) [inline queue.go]: `fmt.Println("Test concurrent nowait Put/Get...if deadlock block forever or panic!")`
`go func() {
		defer wg.Done()
		for {
			select {
			case <-putDone:
				return
			default:
				_, _ = queue.GetNoWait()
			}
		}
	}()
	go func() {
		defer wg.Done()
		time.Sleep(3 * time.Second)
		for i := 0; i < 100; i++ {
			_ = queue.PutNoWait(1)
		}
		putDone <- true
	}()`
Concurrent run the two goroutines if not add the unlock operation the process may be block and deadlock or panic....

Comment (damnever) [inline queue.go]: FYI:
> New create a new Queue, The maxSize variable sets the max Queue size. If maxSize is zero, Queue will be infinite size, and Put always no wait.

Comment (Arvintian) [inline queue.go]: But your every put/get operation have lock operation and 
```
	if timeout < 0.0 && isempty {
		return nil, ErrEmptyQueue
	}
```

this exit case not unlock.

Comment (damnever) [inline queue.go]: Yes, that a bug, thank you to fix it, but your test case does not cover the code that you added.

Comment (Arvintian) [inline queue.go]: I just want to simulate the scene where I found the bug yet the test case I writed is poor . you fix it and I will close the pr.

**Comment by damnever** [comment]
@Arvintian Ok, I will fix it as soon as I get time. Thanks for your PR anyway.

**Pull request closed without merging**